common-arcana.lic

```diff
@@ -640,4 +640,23 @@ module DRCA
 
     crafting_prepare_spell(training_spells[needs_training], settings)
   end
+
+  def cast_waggle(settings, waggle_sets_name = 'default')
+    return unless settings.waggle_sets[waggle_sets_name]
+
+    spells = settings.waggle_sets[waggle_sets_name]
+
+    if DRStats.barbarian?
+      start_barb_abilities(spells, settings)
+    elsif DRStats.thief?
+      start_khris(spells, settings)
+    else
+      spells.each_value do |value|
```

Comment:
💡 Could we do something like this?

```ruby
spells.values
.select { |spell| spell['use_auto_mana'] }
.each { |spell| check_discern(spell, settings) }
```

Reply:
I think it should work -- I'll try it out.